test/gtest/uct/test_peer_failure.cc

```diff
@@ -43,28 +49,38 @@ class test_uct_peer_failure : public uct_test {
         reinterpret_cast<test_uct_peer_failure*>(arg)->m_err_count++;
     }
 
-    void kill_receiver0()
+    void kill_receiver()
     {
-        m_entities.remove(m_receiver1);
-        ucs_assert(m_entities.size() == 2);
+        ucs_assert(!m_receivers.empty());
+        m_entities.remove(m_receivers.front());
+        ucs_assert(m_entities.size() == m_receivers.size());
+        m_receivers.pop();
     }
 
     void set_am_handlers()
     {
         check_caps(UCT_IFACE_FLAG_AM_CB_SYNC);
-        uct_iface_set_am_handler(m_receiver1->iface(), 0, am_dummy_handler,
-                                 reinterpret_cast<void*>(this), UCT_AM_CB_FLAG_SYNC);
-        uct_iface_set_am_handler(m_receiver2->iface(), 0, am_dummy_handler,
-                                 reinterpret_cast<void*>(this), UCT_AM_CB_FLAG_SYNC);
+
+        std::queue<entity *> tmp;
+        entity *e;
+        while (!m_receivers.empty()) {
```

Comment:
pls use "for" loop to iterate over STL container
http://www.cprogramming.com/tutorial/stl/iterators.html